src/cryptography/hazmat/backends/openssl/ciphers.py

```diff
@@ -131,8 +133,10 @@ def update_into(self, data, buf):
             "unsigned char *", self._backend._ffi.from_buffer(buf)
         )
         outlen = self._backend._ffi.new("int *")
-        res = self._backend._lib.EVP_CipherUpdate(self._ctx, buf, outlen,
-                                                  data, len(data))
+        res = self._backend._lib.EVP_CipherUpdate(
+            self._ctx, buf, outlen,
+            self._backend._ffi.from_buffer(data), len(data)
```

Comment:
I'm sure I'll regret asking, but what happens if you pass the same buffer for `data` and `buf`?

Reply:
It will fail with a ValueError because we force buffer size to be `block_size - 1` larger than the data passed so that we can always write everything back to the supplied buffer. If we were to remove that restriction (which I am not proposing we do since it exists to prevent segfaults in cases where users are not careful about block alignment) then it would actually properly do an in place update from plaintext to ciphertext (and vice versa on decrypt).